rust/libnewsboat/src/fmtstrformatter/mod.rs

```diff
@@ -190,6 +210,14 @@ impl FmtStrFormatter {
 mod tests {
     use super::*;
 
+    #[test]
+    fn t_do_format_center() {
+        let mut fmt = FmtStrFormatter::new();
+        fmt.register_fmt('T', "whatever".to_string());
+        assert_eq!(fmt.do_format("%=20T", 0), "      whatever      ");
+        assert_eq!(fmt.do_format("%=19T", 0), "     whatever      ");
```

Comment:
It'd be nice to have tests for cases where padding width is less than the value's length, e.g. `%=3T` where `T` is `"whatever"`. Don't forget to add the same to the C++ test.

Reply:
I added the test and actually fixed it to pass the test as well